rollouts/controllers/rollout_controller.go

```diff
@@ -294,30 +307,16 @@ func (r *RolloutReconciler) reconcileRollout(ctx context.Context, rollout *gitop
 
 	waveStatuses := []gitopsv1alpha1.WaveStatus{}
 
-	for _, wave := range strategy.Spec.Waves {
-		waveClusters, err := r.store.ListClusters(ctx, rollout.Spec.Targets.Selector, wave.Targets.Selector)
-		if err != nil {
-			return err
-		}
-
-		packageClusterMatcherClient := packageclustermatcher.NewPackageClusterMatcher(waveClusters.Items, discoveredPackages)
-		allClusterPackages, err := packageClusterMatcherClient.GetClusterPackages(rollout.Spec.PackageToTargetMatcher)
-		if err != nil {
-			logger.Error(err, "get cluster packages failed")
-			return client.IgnoreNotFound(err)
-		}
-
-		for _, clusterPackages := range allClusterPackages {
-			clusterName := clusterPackages.Cluster.Name
-			logger.Info("cluster packages", "cluster", clusterName, "packagesCount", len(clusterPackages.Packages), "packages", clusterPackages.Packages)
-		}
+	allWaveTargets, err := r.getWaveTargets(ctx, rollout, targets, strategy.Spec.Waves)
+	if err != nil {
+		return err
+	}
 
-		targets, err := r.computeTargets(ctx, rollout, allClusterPackages, waveClusters.Items)
-		if err != nil {
-			return err
-		}
+	for _, waveTargets := range allWaveTargets {
```

Comment:
nit: can we move `pauseFutureWaves` and other waves variable closer to this loop. I think of this phase as `rolloutWaves` so all state related to rolling out waves would be good to have in one section. And potentially we can move this in a function later.

Reply:
Agreed. Updated.